crates/ruff_server/src/format.rs

```diff
@@ -3,10 +3,10 @@ use ruff_python_formatter::format_module_source;
 use ruff_text_size::TextRange;
 use ruff_workspace::FormatterSettings;
 
-use crate::edit::Document;
+use crate::edit::TextDocument;
 
 pub(crate) fn format(
-    document: &Document,
+    document: &TextDocument,
     formatter_settings: &FormatterSettings,
 ) -> crate::Result<String> {
     // TODO(jane): support Jupyter Notebook
```

Comment:
I believe this comment is not needed now? Also, you'd need to use the `source_type` here to determine the parser mode because otherwise it errors for a notebook containing escape commands.